### PR TITLE
Modular auth backend

### LINKED ISSUE #1148: Refactor auth backend to make it dynamic
There's currently limited support for [auth](https://github.com/Screenly/screenly-ose/blob/master/docs/http-basic-authentication.md) in Screenly OSE. It would be nice to make this more modular such that we can add more backends in the future. 

As such, let's refactor the way that auth is done right now and do the following:

 * Introduce a new settings in the dashboard where a user can pick the auth backend.
 * Refactor the auth code (such as [this](https://github.com/Screenly/screenly-ose/blob/master/settings.py#L45-L46) to make it modular and support multiple backends.
 * Refactor the auth checks to pick the auth backend from a given system (currently only the bult-in, but design it to support multiple ones).

### PR DESCRIPTION
**Pull request by a-martynovich**
Closes #1148

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -70,6 +64,12 @@ def __init__(self, *args, **kwargs):
         IterableUserDict.__init__(self, *args, **kwargs)
         self.home = getenv('HOME')
         self.conf_file = self.get_configfile()
+        self.auth_backends_list = [NoAuth(), BasicAuth(self), WoTTAuth(self)]
+        self.auth_backends = {}
+        for b in self.auth_backends_list:
+            c = b.config
```
Comment (korvyashka) [inline settings.py]: Minor : b and c are not readable names.

File: auth.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    @property
+    def is_authorized(self):
+        auth = request.authorization
+        return auth and self._check(auth.username, auth.password)
+
+    @property
+    def template(self):
+        return 'auth_basic.html', {'user': self.settings['user']}
+
+    def authenticate(self):
```
[the remark this answers is not in the document]
Comment (a-martynovich) [reply auth.py]: I agree with your statement about terminology. I’ll come up with something better.
However the function does exactly what the docstring says: initiates authentication. Okay, the browser initiates it as a result of getting 401. But if we will have another auth backend which can generate its own login page that’s what this function will return. And that will truly initiate authentication.

[the remark this answers is not in the document]
Comment (korvyashka) [reply auth.py]: But looking at the authorize function above it seems like authenticate is used to always return fail response.
```
if not self.is_authorized:
   return self.authenticate()
```

**Comment by a-martynovich** [comment]
Closed in favor of #1165

**Pull request closed without merging**